src/assets/js/alma-checkout-in-page.js

```diff
@@ -0,0 +1,167 @@
+/**
+ * Checkout page.
+ *
+ * @since 5.0.0
+ *
+ * @param global alma_iframe_params iframe params.
+ *
+ * @package Alma_Gateway_For_Woocommerce
+ */
+
+(function ($) {
+	var regex  = /^general_([0-9]+)_[0-9]+_[0-9]+/g;
+	var inPage = undefined;
+
+	$( 'body' ).on(
+		'updated_checkout',
+		function (event) {
+			if (isAlmaInPageChecked()) {
+				feePlanChecked = $( "input[type='radio'][name='alma_fee_plan_in_page']:checked" ).val();
+				render_installments( feePlanChecked );
+			}
+		}
+	);
+
+	$( 'body' ).on(
+		'click change',
+		'input[name="payment_method"]',
+		function (event) {
+			if (isAlmaInPageChecked()) {
+				feePlanChecked = $( "input[type='radio'][name='alma_fee_plan_in_page']:checked" ).val();
+				render_installments( feePlanChecked );
+			}
+		}
+	);
+
+	$( 'body' ).on(
+		'click',
+		'.alma_fee_plan_in_page',
+		function (event) {
+			render_installments( $( this ).prop( 'value' ) );
+		}
+	);
+
+	function render_installments(value)
+	{
+		payment_method = $( '.woocommerce-checkout input[name="payment_method"]:checked' );
+		payment_value  = payment_method.attr( 'value' );
+
+		matches = value.matchAll( regex );
+
+		installment = null;
+
+		for (const match of matches) {
+			installment = match[1];
+		}
+
+		if (null !== installment) {
+			if ( inPage !== undefined) {
+				inPage.unmount();
+			}
+
+			inPage = Alma.InPage.initialize(
+				{
+					merchantId: alma_iframe_params.merchant_id,
+					amountInCents: alma_iframe_params.amount_in_cents,
+					installmentsCount: installment,
+					selector: "#alma-inpage-" + payment_value,
+					environment: alma_iframe_params.environment,
+					locale: alma_iframe_params.locale,
+				}
+			);
+		}
+	}
+
+	function isAlmaInPageChecked() {
+		return $( '#payment_method_alma_in_page' ).is( ':checked' ) || $( '#payment_method_alma_in_page_pay_now' ).is( ':checked' );
+	}
+
+	$( 'body' ).on(
+		'click',
+		'button[name="woocommerce_checkout_place_order"]',
+		function(evt) {
+
+			if (isAlmaInPageChecked()) {
+				evt.preventDefault();
+				feePlanChecked = $( "input[type='radio'][name='alma_fee_plan_in_page']:checked" ).val();
+
+				var data = {
+					'action': 'alma_do_checkout_in_page',
+					'fields': $( 'form.checkout' ).serializeArray(),
+					'alma_fee_plan_in_page': feePlanChecked
+				};
+
+				// Create the payment id and order.
+				jQuery.post( ajax_object.ajax_url, data )
+					.done(
+						function (response) {
+							var dataReturn = {
+								'action': 'alma_return_checkout_in_page',
+								'alma_fee_plan_in_page': feePlanChecked
+							};
+
+							paymentId = response.data.payment_id;
+							orderId   = response.data.order_id;
+
+							// Start the payment.
+							inPage.startPayment(
+								{
+									paymentId:paymentId,
+									onUserCloseModal: () => {
+										cancel_order( orderId )
+									}
+								}
+							);
+
+						}
+					)
+				.fail(
+					function (response) {
+						location.reload();
```

Comment:
Do you have an error message after reload?

Reply:
Yes, it display the woocommerce errors message i send like in helpers/class-alma-order-helper.php line 270

Reply:
done